sc-poc/modules/pubsub/asyncapi.go

```diff
@@ -84,29 +84,51 @@ func (a *App) generateASyncAPIDoc() *AsyncAPI {
 			Subscribe: &Operation{
 				ID: "producerSubscribe" + getID(channelObj.Name),
 				Message: MessageOneOrMany{
-					MessageEntity: MessageEntity{
-						Name:        "Acknowledgement",
-						ContentType: "application/json",
-						Payload: map[string]interface{}{
-							"type": "object",
-							"properties": map[string]interface{}{
-								"event": map[string]interface{}{
-									"type": "string",
+					OneOf: []MessageEntity{
+						{
+							Name:        "Acknowledgement",
+							ContentType: "application/json",
+							Payload: map[string]interface{}{
+								"type": "object",
+								"properties": map[string]interface{}{
+									"event": map[string]interface{}{
+										"type": "string",
+									},
+									"data": map[string]interface{}{
+										"type": "object",
+										"properties": map[string]interface{}{
+											"id": map[string]interface{}{
+												"type": "string",
+											},
+											"ack": map[string]interface{}{
+												"type": "boolean",
+											},
+										},
+										"required": []string{"id", "ack"},
+									},
 								},
-								"data": map[string]interface{}{
-									"type": "object",
-									"properties": map[string]interface{}{
-										"id": map[string]interface{}{
+								"required": []string{"event", "data"},
+							},
+						},
+						{
+							Name:        "Error",
+							ContentType: "application/json",
+							Payload: map[string]interface{}{
+								"type": "object",
+								"properties": map[string]interface{}{
+									"message": map[string]interface{}{
```

Comment:
Accept a message id as well. We need to tell the publisher which message didn't publish. Also you can make json schema directly from a struct without having to make such maps. Look at this library: https://github.com/invopop/jsonschema

Reply:
Can we do this in a separate PR? I will make an issue for it.

Reply:
Ok

Reply:
https://github.com/spacecloud-io/space-cloud/issues/1680